packages/lib-utils/src/components/resource-summary/resource-summary.test.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import type { K8sResourceCommon } from '../../index';
+import { ResourceSummary } from './index';
+
+expect.extend(toHaveNoViolations);
+
+const k8Object: K8sResourceCommon = {
+  apiVersion: 'v1beta1',
+  apiGroup: 'tenancy.kcp.dev',
+  kind: 'Workspace',
+  metadata: {
+    name: 'demo-workspace',
+    labels: {
+      label1: 'value1',
+      label2: 'value2',
+    },
+    creationTimestamp: '2022-09-15T21:07:32Z',
+    ownerReferences: [
+      {
+        name: 'owner1',
+        apiVersion: 'apps/v1',
+        kind: 'StatefulSet',
+        uid: '8f5ce882-4a20-47da-ad6a-16d0c4846405',
+      },
+      {
+        name: 'owner2',
+        apiVersion: 'apps/v1',
+        kind: 'StatefulSet',
+        uid: '8f5ce882-4a20-47da-ad6a-16d0c4846406',
+      },
+    ],
+  },
+};
+
+describe('ResourceSummary', () => {
+  it('is accessible', async () => {
+    const { container } = render(<ResourceSummary resource={k8Object} />);
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  it('is accessible - loading', async () => {
+    const { container } = render(<ResourceSummary loaded={false} />);
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  it('displays data', () => {
+    const { container } = render(
+      <ResourceSummary resource={k8Object}>
+        <dt>My custom label</dt>
+        <dd>My custom value</dd>
+      </ResourceSummary>,
+    );
+    expect(screen.getByText('demo-workspace')).toBeInTheDocument();
+    expect(screen.getByText('label1=value1')).toBeInTheDocument();
+    expect(screen.getByText('2022-09-15T21:07:32Z')).toBeInTheDocument();
+    expect(screen.getByText('owner1')).toBeInTheDocument();
+    expect(screen.getByText('owner2')).toBeInTheDocument();
+
+    expect(screen.getByText('My custom label')).toBeInTheDocument();
+    expect(screen.getByText('My custom value')).toBeInTheDocument();
+
+    expect(screen.queryByText('Namespace')).not.toBeInTheDocument();
+    expect(screen.queryByText('No owner')).not.toBeInTheDocument();
+    expect(container.querySelectorAll('.pf-c-skeleton').length).toEqual(0);
+  });
+
+  it('displays with missing owner', () => {
+    const k8Object2: K8sResourceCommon = {
+      apiVersion: 'v1beta1',
+      apiGroup: 'tenancy.kcp.dev',
+      kind: 'Workspace',
+      metadata: {
+        name: 'demo-workspace',
+      },
+    };
+
+    render(<ResourceSummary resource={k8Object2} />);
+    expect(screen.getByText('No owner')).toBeInTheDocument();
+  });
+
+  it('displays skeletons when loading', () => {
+    const k8Object2: K8sResourceCommon = {
+      apiVersion: 'v1beta1',
+      apiGroup: 'tenancy.kcp.dev',
+      kind: 'Workspace',
+      metadata: {
+        name: 'demo-workspace',
+      },
+    };
+
+    const { container } = render(<ResourceSummary resource={k8Object2} loaded={false} />);
```

Comment:
@kdoberst wondering if we could utilize lodash `omit` on the `ownerReferences` key so we don't need separate objects for this: https://lodash.com/docs/4.17.15#omit

Reply:
I'm honestly not a fan lodash because it adds overhead that isn't needed in most cases (unless you are doing full-on functional programming).  I'm also a fan of having completely separate objects for testing vs modifying the data.  Makes the tests easier to read and modify without worrying about changing the data messing up other tests.

I can change it if you like

Reply:
Chatted with Kim - not blocking. Mainly a maintenance concern in the future (3 places we may need to change a key/value over one). That could in theory be a good thing. For now this is fine as is.